apps/site/assets/ts/schedule/components/direction/ScheduleDirectionMenu.tsx

```diff
@@ -1,44 +1,359 @@
+      routePattern.id !== compareRoutePattern.id &&
+      routePattern.headsign === compareRoutePattern.headsign
+  ).length > 0;
+
+const ExpandedMenu = ({
+  routePatterns,
+  selectedRoutePatternId,
+  showAllRoutePatterns,
+  itemFocus,
+  dispatch
+}: ExpandedMenuProps): ReactElement<HTMLElement> => {
+  const filterRule = (routePattern: EnhancedRoutePattern): boolean =>
+    showAllRoutePatterns ? true : routePattern.typicality < 3;
+  const handleClick = (): void =>
+    dispatch({ event: "showAllRoutePatterns", payload: {} });
+  const focusIndex = determineFocusIndex(itemFocus, routePatterns);
+  const toggleButtonIds =
+    hasMoreRoutePatterns(routePatterns) && showAllRoutePatterns === false
+      ? ["uncommon"]
+      : [];
+  const routePatternIds = routePatterns
+    .filter(filterRule)
+    .map((routePattern: EnhancedRoutePattern) => routePattern.id)
+    .concat(toggleButtonIds);
+  return (
+    <div className="m-schedule-direction__menu" role="menu">
+      {routePatterns
+        .filter(filterRule)
+        .map((routePattern: EnhancedRoutePattern, index: number) => (
+          <RoutePatternItem
+            key={routePattern.id}
+            routePatternIds={routePatternIds}
+            routePattern={routePattern}
+            selected={selectedRoutePatternId === routePattern.id}
+            focused={index === focusIndex}
+            duplicated={isDuplicateHeadsign(routePattern, routePatterns)}
+            dispatch={dispatch}
+          />
+        ))}
+      {hasMoreRoutePatterns(routePatterns) && showAllRoutePatterns === false && (
+        <div
+          id="route-pattern_uncommon"
+          role="menuitem"
+          tabIndex={0}
+          className="m-schedule-direction__menu-item"
+          onClick={handleClick}
+          onKeyUp={(e: ReactKeyboardEvent) =>
+            handleReactEnterKeyPress(e, () => {
+              handleClick();
+            })
+          }
+          onKeyDown={(e: ReactKeyboardEvent) => {
+            handleNavigation(e, routePatternIds);
+          }}
+        >
+          <div className="m-schedule-direction__menu-item-more">
+            Uncommon destinations{" "}
+            {renderSvg(
+              "c-svg__icon m-schedule-direction__route-pattern-arrow",
+              arrowIcon
+            )}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+const routePatternNameById = (
+  routePatterns: EnhancedRoutePattern[],
+  selectedRoutePatternId: string
+): string =>
+  routePatterns.find(
+    (routePattern: EnhancedRoutePattern) =>
+      routePattern.id === selectedRoutePatternId
+  )!.headsign;
+
+/* istanbul ignore next */
+const externalCloseEvent = (dispatch: Dispatch<Action>): void => {
+  document.addEventListener(
+    "click",
+    (event: Event): void => {
+      if (!event.target) return;
+      const element = event.target as HTMLElement;
+      if (element.closest(".js-m-schedule-click-boundary")) return;
```

Comment:
Closest isnt supported in IE11. Do we have or can we add a polyfill?

https://caniuse.com/#search=closest